### PR TITLE
fix: try sending unsolicited reports to set the time/date via `Time CC`

### LINKED ISSUE #6032: Send unsolicited TimeCC report when setting the time on a device that only supports Time CC
### Is your problem within Home Assistant (Core or Z-Wave JS Integration)?

NO, my problem is NOT within Home Assistant or the ZWave JS integration

### Is your problem within Z-Wave JS UI (formerly ZwaveJS2MQTT)?

NO, my problem is NOT within Z-Wave JS UI

### Checklist

- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there.

### Describe the bug

What causes the bug?
Unknown

What do you observe?
The clock on the lock never gets successfully set.  I can query the lock for the time and date and it's always wrong.

```
2023-07-18T22:29:34.387Z DRIVER » [Node 087] [REQ] [SendData]
                                  │ transmit options: 0x25
                                  │ callback id:      63
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 168
                                    │ security class:  S2_AccessControl
                                    └─[TimeCCTimeGet]

2023-07-18T22:29:35.708Z DRIVER « [Node 087] [REQ] [ApplicationCommand]
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 95
                                    │ security class:  S2_AccessControl
                                    └─[TimeCCTimeReport]
                                        time: 00:05:33



2023-07-18T22:29:35.744Z DRIVER » [Node 087] [REQ] [SendData]
                                  │ transmit options: 0x25
                                  │ callback id:      64
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 169
                                    │ security class:  S2_AccessControl
                                    └─[TimeCCDateGet]

2023-07-18T22:29:35.818Z DRIVER « [Node 087] [REQ] [ApplicationCommand]
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 96
                                    │ security class:  S2_AccessControl
                                    └─[TimeCCDateReport]
                                        date: 1970-01-01
```
(full log attached)

Pulling the battery on the lock and reinserting it will trigger the lock to ask for the time, but it setting it seems to fail.

```
2023-07-18T22:24:09.045Z DRIVER « [Node 087] [REQ] [ApplicationCommand]
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 90
                                    │ security class:  S2_AccessControl
                                    └─[SupervisionCCGet]
                                      │ session id:      2
                                      │ request updates: false
                                      └─[TimeCCTimeGet]

2023-07-18T22:24:09.066Z DRIVER » [Node 087] [REQ] [SendData]
                                  │ transmit options: 0x25
                                  │ callback id:      55
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 163
                                    │ security class:  S2_AccessControl
                                    └─[SupervisionCCGet]
                                      │ session id:      62
                                      │ request updates: true
                                      └─[TimeCCTimeReport]
                                          time: 17:24:09




2023-07-18T22:24:09.117Z DRIVER « [Node 087] [REQ] [ApplicationCommand]
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 91
                                    │ security class:  S2_AccessControl
                                    └─[SupervisionCCReport]
                                        session id:          62
                                        more updates follow: false
                                        status:              Fail
                                        duration:            0s

2023-07-18T22:24:09.144Z DRIVER » [Node 087] [REQ] [SendData]
                                  │ transmit options: 0x24
                                  │ callback id:      0
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 164
                                    │ security class:  S2_AccessControl
                                    └─[SupervisionCCReport]
                                        session id:          2
                                        more updates follow: false
                                        status:              Success
                                        duration:            0s
```
(full log attached)

What did you expect to happen?
The time to get set correctly.

I half expect this to be a firmware problem as this works with another lock I have using S2 Access Control and Time CC v1, but I don't know what I'm looking at enough to be sure.

### Device information

Manufacturer: Schlage
Model name:  Keypad Lever (JFE109)
Node ID in your network: 87

[node_87.json.txt](https://github.com/zwave-js/node-zwave-js/files/12089479/node_87.json.txt)

### How are you using `node-zwave-js`?

- [X] `zwave-js-ui` (formerly `zwavejs2mqtt`) Docker image (latest)
- [ ] `zwave-js-ui` (formerly `zwavejs2mqtt`) Docker image (dev)
- [ ] `zwave-js-ui` (formerly `zwavejs2mqtt`) Docker manually built _(please specify branches)_
- [ ] `ioBroker.zwave2` adapter _(please specify version)_
- [ ] `HomeAssistant zwave_js` integration  _(please specify version)_
- [ ] `pkg`
- [ ] `node-red-contrib-zwave-js` _(please specify version, double click node to find out)_
- [ ] Manually built from GitHub _(please specify branch)_
- [ ] Other _(please describe)_

### Which branches or versions?

version:
zwave-js-ui: 8.19.0
zwave-js: 11.0.0

### Did you change anything?

no

### If yes, what did you change?

_No response_

### Did this work before?

No, it never worked anywhere

### If yes, where did it work?

Not to my knowledge, at least.

### Attach Driver Logfile

[zwavejs_2023-07-18.log](https://github.com/zwave-js/node-zwave-js/files/12089482/zwavejs_2023-07-18.log)

### PR DESCRIPTION
**Pull request by AlCalzone**
fixes: #6032

@kdknigga this needs a test from you since I don't have any devices that only support Time CC.
Can you do that from a test build, e.g. with a [custom Docker container](https://zwave-js.github.io/zwave-js-ui/#/development/custom-docker?id=building-a-container-using-dockerfilecontrib)?
Otherwise we can just iterate after release.

### CODE REVIEW

**Comment by kdknigga** [comment]
I'll be able to give it a go tomorrow.

**Comment by kdknigga** [comment]
With the test build, I'm seeing versions:

zwave-js-ui: 8.21.0.1b7eadb
zwave-js: 11.5.4-0.dev-7247a0ff9

Does that look correct?

If so...


I'm tailing both the ZUI and zwavejs logs while I hit the "sync" button:

```
2023-07-21T19:08:36.512Z DRIVER   all queues idle

==> /home/kknigga/.local/share/containers/storage/volumes/zwavejs2mqtt/_data/logs/z-ui_current.log <==
2023-07-21 19:08:52.286 DEBUG SOCKET: Event ZWAVE_API emitted to t3tGMK-fvrVgYE6lAAAD
2023-07-21 19:08:52.286 INFO Z-WAVE: Calling api syncNodeDateAndTime with args: [ 9, [length]: 1 ]
2023-07-21 19:08:52.287 INFO Z-WAVE: [Node 009] Syncing Node 9 date and time
2023-07-21 19:08:52.288 INFO Z-WAVE: Success zwave api call syncNodeDateAndTime false

==> /home/kknigga/.local/share/containers/storage/volumes/zwavejs2mqtt/_data/logs/zwavejs_current.log <==
2023-07-21T19:09:06.504Z DRIVER   all queues busy
2023-07-21T19:09:06.505Z SERIAL » 0x0103003bc7                                                         (5 bytes)
2023-07-21T19:09:06.506Z DRIVER » [REQ] [GetBackgroundRSSI]
```

So, that doesn't look to have worked.

For comparison, syncing a node supporting Clock:

```
2023-07-21T19:13:06.512Z DRIVER   all queues idle

==> /home/kknigga/.local/share/containers/storage/volumes/zwavejs2mqtt/_data/logs/z-ui_current.log <==
2023-07-21 19:13:10.830 DEBUG SOCKET: Event ZWAVE_API emitted to t3tGMK-fvrVgYE6lAAAD
2023-07-21 19:13:10.831 INFO Z-WAVE: Calling api syncNodeDateAndTime with args: [ 6, [length]: 1 ]
2023-07-21 19:13:10.831 INFO Z-WAVE: [Node 006] Syncing Node 6 date and time
2023-07-21 19:13:10.885 INFO Z-WAVE: Success zwave api call syncNodeDateAndTime true

==> /home/kknigga/.local/share/containers/storage/volumes/zwavejs2mqtt/_data/logs/zwavejs_current.log <==
2023-07-21T19:13:10.834Z DRIVER   all queues busy
2023-07-21T19:13:10.836Z SERIAL » 0x011000a90106048104b30d25000000002279                              (18 bytes)
2023-07-21T19:13:10.837Z DRIVER » [Node 006] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      34
                                  └─[ClockCCSet]
                                      clock setting: Friday, 19:13
2023-07-21T19:13:10.841Z SERIAL « [ACK]                                                                   (0x06)
2023-07-21T19:13:10.843Z SERIAL « 0x010401a90152                                                       (6 bytes)
2023-07-21T19:13:10.843Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T19:13:10.843Z DRIVER « [RES] [SendDataBridge]
                                    was sent: true
2023-07-21T19:13:10.880Z SERIAL « 0x011d00a92200000301a87f7f7f7f00000303000000030100007f7f7f7f7fbe    (31 bytes)
2023-07-21T19:13:10.880Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T19:13:10.880Z DRIVER « [REQ] [SendDataBridge]
                                    callback id:            34
                                    transmit status:        OK, took 30 ms
                                    repeater node IDs:      3
                                    routing attempts:       1
                                    protocol & route speed: Z-Wave, 100 kbit/s
                                    ACK RSSI:               -88 dBm
                                    ACK RSSI on repeaters:  N/A
                                    ACK channel no.:        0
                                    TX channel no.:         0
2023-07-21T19:13:10.885Z DRIVER   all queues idle
```

File: packages/zwave-js/src/lib/node/Node.ts [hunk 1]
```diff
@@ -5896,9 +5896,72 @@ ${formatRouteHealthCheckSummary(this.id, otherNode.id, summary)}`,
 			} catch {
 				return false;
 			}
-		} else {
-			// No way to set the time
-			return false;
+		} else if (
+			timeAPI.isSupported() &&
+			timeAPI.supportsCommand(TimeCommand.DateReport) &&
+			timeAPI.supportsCommand(TimeCommand.TimeReport)
```
Comment (kdknigga) [inline packages/zwave-js/src/lib/node/Node.ts:5902]: Would there ever be a case where Time CC is supported, but these two commands aren't?  It looks like they're required in both versions of Time CC.

**Comment by kdknigga** [comment]
Ok, turns out I did build the container wrong.  Here's attempt number 2.

version 11.5.3-0.dev-20166b6fe

```
==> /home/kknigga/.local/share/containers/storage/volumes/zwavejs2mqtt/_data/logs/z-ui_current.log <==                                                                                                                                        
2023-07-21 22:05:14.328 DEBUG SOCKET: Event ZWAVE_API emitted to vJStjYSR2mZQoh7mAAAD                                                                                                                                                         
2023-07-21 22:05:14.328 INFO Z-WAVE: Calling api syncNodeDateAndTime with args: [ 9, [length]: 1 ]                                                                                                                                            
2023-07-21 22:05:14.329 INFO Z-WAVE: [Node 009] Syncing Node 9 date and time                                                                                                                                                                  
                                                                                                                                                                                                                                              
==> /home/kknigga/.local/share/containers/storage/volumes/zwavejs2mqtt/_data/logs/zwavejs_current.log <==                                                                                                                                     
2023-07-21T22:05:14.330Z DRIVER   all queues busy
2023-07-21T22:05:14.332Z SERIAL » 0x011e00a90109129f03bb0077322e22c7b81ef3645046bd4fc725000000000dc1  (32 bytes)
2023-07-21T22:05:14.333Z DRIVER » [Node 009] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      13
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 187
                                    │ security class:  S2_AccessControl
                                    └─[TimeCCDateReport]
                                        date: 2023-07-21
2023-07-21T22:05:14.340Z SERIAL « [ACK]                                                                   (0x06)
2023-07-21T22:05:14.342Z SERIAL « 0x010401a90152                                                       (6 bytes)
2023-07-21T22:05:14.342Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:14.343Z DRIVER « [RES] [SendDataBridge]
                                    was sent: true
2023-07-21T22:05:15.575Z SERIAL « 0x011d00a90d00007b00aa7f7f7f7f01010300000000420100007f7f7f7f7fa8    (31 bytes)
2023-07-21T22:05:15.576Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:15.577Z DRIVER « [REQ] [SendDataBridge]
                                    callback id:            13
                                    transmit status:        OK, took 1230 ms
                                    routing attempts:       1
                                    protocol & route speed: Z-Wave, 40 kbit/s
                                    ACK RSSI:               -86 dBm
                                    ACK channel no.:        1
                                    TX channel no.:         1
                                    beam:                   1000 ms
2023-07-21T22:05:16.083Z SERIAL » 0x011a00a901090e9f03bc00d8d95e5bcc9cf2e2a07325000000000ed6          (28 bytes)
2023-07-21T22:05:16.084Z DRIVER » [Node 009] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      14
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 188
                                    │ security class:  S2_AccessControl
                                    └─[TimeCCDateGet]
2023-07-21T22:05:16.089Z SERIAL « [ACK]                                                                   (0x06)
2023-07-21T22:05:16.091Z SERIAL « 0x010401a90152                                                       (6 bytes)
2023-07-21T22:05:16.091Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:16.092Z DRIVER « [RES] [SendDataBridge]
                                    was sent: true
2023-07-21T22:05:16.109Z SERIAL « 0x011d00a90e00000100aa7f7f7f7f01010300000000020100007f7f7f7f7f91    (31 bytes)
2023-07-21T22:05:16.109Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:16.110Z DRIVER « [REQ] [SendDataBridge]
                                    callback id:            14
                                    transmit status:        OK, took 10 ms
                                    routing attempts:       1
                                    protocol & route speed: Z-Wave, 40 kbit/s
                                    ACK RSSI:               -86 dBm
                                    ACK channel no.:        1
                                    TX channel no.:         1
2023-07-21T22:05:16.133Z SERIAL « 0x011b00a8000109129f03bc00f02ba8ecb2f2254469f0799d9a8500aa00        (29 bytes)
2023-07-21T22:05:16.134Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:16.135Z DRIVER « [Node 009] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -86 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 188
                                    │ security class:  S2_AccessControl
                                    └─[TimeCCDateReport]
                                        date: 2023-07-21
2023-07-21T22:05:16.137Z SERIAL » 0x011d00a90109119f03bd00d4ecd35a702dee7e768f78fc1425000000000f4c    (31 bytes)
2023-07-21T22:05:16.138Z DRIVER » [Node 009] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      15
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 189
                                    │ security class:  S2_AccessControl
                                    └─[TimeCCTimeReport]
                                        time: 22:05:14
2023-07-21T22:05:16.143Z SERIAL « [ACK]                                                                   (0x06)
2023-07-21T22:05:16.145Z SERIAL « 0x010401a90152                                                       (6 bytes)
2023-07-21T22:05:16.145Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:16.146Z DRIVER « [RES] [SendDataBridge]
                                    was sent: true
2023-07-21T22:05:16.163Z SERIAL « 0x011d00a90f00000100aa7f7f7f7f01010300000000020100007f7f7f7f7f90    (31 bytes)
2023-07-21T22:05:16.163Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:16.164Z DRIVER « [REQ] [SendDataBridge]
                                    callback id:            15
                                    transmit status:        OK, took 10 ms
                                    routing attempts:       1
                                    protocol & route speed: Z-Wave, 40 kbit/s
                                    ACK RSSI:               -86 dBm
                                    ACK channel no.:        1
                                    TX channel no.:         1

==> /home/kknigga/.local/share/containers/storage/volumes/zwavejs2mqtt/_data/logs/z-ui_current.log <==
2023-07-21 22:05:16.764 INFO Z-WAVE: Success zwave api call syncNodeDateAndTime true

==> /home/kknigga/.local/share/containers/storage/volumes/zwavejs2mqtt/_data/logs/zwavejs_current.log <==
2023-07-21T22:05:16.668Z SERIAL » 0x011a00a901090e9f03be0025c877542c879cbee51b250000000010e4          (28 bytes)
2023-07-21T22:05:16.669Z DRIVER » [Node 009] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      16
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 190
                                    │ security class:  S2_AccessControl
                                    └─[TimeCCTimeGet]
2023-07-21T22:05:16.674Z SERIAL « [ACK]                                                                   (0x06)
2023-07-21T22:05:16.676Z SERIAL « 0x010401a90152                                                       (6 bytes)
2023-07-21T22:05:16.676Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:16.677Z DRIVER « [RES] [SendDataBridge]
                                    was sent: true
2023-07-21T22:05:16.692Z SERIAL « 0x011d00a91000000100aa7f7f7f7f01010300000000020100007f7f7f7f7f8f    (31 bytes)
2023-07-21T22:05:16.693Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:16.693Z DRIVER « [REQ] [SendDataBridge]
                                    callback id:            16
                                    transmit status:        OK, took 10 ms
                                    routing attempts:       1
                                    protocol & route speed: Z-Wave, 40 kbit/s
                                    ACK RSSI:               -86 dBm
                                    ACK channel no.:        1
                                    TX channel no.:         1
2023-07-21T22:05:16.718Z SERIAL « 0x011a00a8000109119f03bd0008c05fce97b4acf42fb4e826a600aa0e          (28 bytes)
2023-07-21T22:05:16.720Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:16.721Z DRIVER « [Node 009] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -86 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 189
                                    │ security class:  S2_AccessControl
                                    └─[TimeCCTimeReport]
                                        time: 22:05:14
2023-07-21T22:05:16.725Z SERIAL » 0x012100a90109159f03bf00cb963664c16a1170c6c49cd7411b5c7a66250000000 (35 bytes)
                                  011eb
2023-07-21T22:05:16.726Z DRIVER » [Node 009] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      17
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 191
                                    │ security class:  S2_AccessControl
                                    └─[SupervisionCCGet]
                                      │ session id:      49
                                      │ request updates: true
                                      └─[ScheduleEntryLockCCTimeOffsetSet]
                                          standard time offset: 0 minutes
                                          DST offset:           0 minutes
2023-07-21T22:05:16.732Z SERIAL « [ACK]                                                                   (0x06)
2023-07-21T22:05:16.734Z SERIAL « 0x010401a90152                                                       (6 bytes)
2023-07-21T22:05:16.734Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:16.735Z DRIVER « [RES] [SendDataBridge]
                                    was sent: true
2023-07-21T22:05:16.751Z SERIAL « 0x011d00a91100000100aa7f7f7f7f01010300000000020100007f7f7f7f7f8e    (31 bytes)
2023-07-21T22:05:16.751Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:16.752Z DRIVER « [REQ] [SendDataBridge]
                                    callback id:            17
                                    transmit status:        OK, took 10 ms
                                    routing attempts:       1
                                    protocol & route speed: Z-Wave, 40 kbit/s
                                    ACK RSSI:               -86 dBm
                                    ACK channel no.:        1
                                    TX channel no.:         1
2023-07-21T22:05:16.761Z SERIAL « 0x011a00a8000109119f03be00a1dbf833ba0f3aeb1659aedd5a00aa6f          (28 bytes)
2023-07-21T22:05:16.762Z SERIAL » [ACK]                                                                   (0x06)
2023-07-21T22:05:16.763Z DRIVER « [Node 009] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -86 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 190
                                    │ security class:  S2_AccessControl
                                    └─[SupervisionCCReport]
                                        session id:          49
                                        more updates follow: false
                                        status:              Success
                                        duration:            0s
2023-07-21T22:05:16.764Z DRIVER   all queues idle
```

Looks good to me.

Somehow this container has lost being in local time again... but that's my problem.

**Review by kdknigga — APPROVED** [review]

**Comment by AlCalzone** [comment]
@zwave-js-bot automerge

**Pull request merged**